pilota-build/src/lib.rs

```diff
@@ -120,7 +121,7 @@ impl Builder<MkProtobufBackend, ProtobufParser> {
 
 impl<MkB, P> Builder<MkB, P>
 where
-    P: Parser,
+    P: Parser + Default,
```

Comment:
Why here need Default trait bound?

Reply:
std::mem::take in line 225 need Default trait bound.

Reply:
Maybe directly transfer the ownership to the build_cx fn?

Reply:
I didn't figure out how to achieve the transfer of ownership. If only transfer parser,  borrow of partially moved value will occur(like this [play ground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ca001959a2df1b4c9e04831cd39b3551)). And because the compile function needs self.mk_backend after build_cx, we also can't transfer the ownership of whole self to build_cx fn.

Reply:
Write it like this? [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=447ba57b917653244cf61c962f6933c1)

Reply:
get it